File: lib/src/state/theme.rs
use schemars::JsonSchema;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Default, Serialize, Deserialize, JsonSchema)]
#[serde(default, rename_all = "camelCase")]
pub struct ThemeCss {
    /// Css Styles for the dock/taskbar
    pub weg: String,
    /// Css Styles for the window manager
    pub toolbar: String,
    /// Css Styles for the window manager
    pub wm: String,
}

#[derive(Debug, Clone, Default, Serialize, Deserialize, JsonSchema)]
#[serde(default, rename_all = "camelCase")]
pub struct ThemeInfo {
    /// Display name of the theme
    pub display_name: String,
    /// Author of the theme
    pub author: String,
    /// Description of the theme
    pub description: String,
    /// Filename of the theme, is overridden by the program on load.
    pub filename: String,
    /// Tags to be used in search
    pub tags: Vec<String>,
}

#[derive(Debug, Clone, Default, Serialize, Deserialize, JsonSchema)]
#[serde(default, rename_all = "camelCase")]
pub struct Theme {
    /// Metadata about the theme
    pub info: ThemeInfo,
    /// Css Styles of the theme
    pub styles: ThemeCss,
}

File: lib/src/state/theme.ts
export interface ThemeCssByApp {
  /** Css Styles for the dock/taskbar */
  weg: string;
  /** Css Styles for the window manager */
  toolbar: string;
  /** Css Styles for the window manager */
  wm: string;
  /** Css Styles for the app launcher */
  launcher: string;
  /** Css Styles for the wall */
  wall: string;
}

export interface ThemeInfo {
  /** Display name of the theme */
  displayName: string;
  /** Author of the theme */
  author: string;
  /** Description of the theme */
  description: string;
  /** Filename of the theme, is overridden by the program on load */
  filename: string;
  /** Tags to be used in search */
  tags: string[];
}

export interface Theme {
  /** Metadata about the theme */
  info: ThemeInfo;
  /** Css Styles of the theme */
  styles: ThemeCssByApp;
}
